### PR TITLE
Ability to pass GPG arguments to publish

### PR DESCRIPTION
**Pull request by joan38**
Not sure if we are ok to break the interface here. If not I can probably add back `gpgPassphrase` and `gpgKeyName` with a `@deprecated` and wire that to `gpgArgs`.

### CODE REVIEW

File: scalalib/src/publish/SonatypePublisher.scala [hunk 1]
```diff
@@ -154,12 +152,9 @@ class SonatypePublisher(uri: String,
...
 
   // http://central.sonatype.org/pages/working-with-pgp-signatures.html#signing-a-file
-  private def poorMansSign(file: os.Path, maybePassphrase: Option[String], maybeKeyName: Option[String]): os.Path = {
+  private def gpgSigned(file: os.Path, args: Seq[String]): os.Path = {
     val fileName = file.toString
-    val optionFlag = (flag: String, ov: Option[String]) => ov.map(flag :: _ :: Nil).getOrElse(Nil)
-    val command = "gpg" ::
-      optionFlag("--passphrase", maybePassphrase) ++ optionFlag("-u", maybeKeyName) ++
-        Seq("--batch", "--yes", "-a", "-b", fileName)
+    val command = Seq("gpg", "--batch", "--yes", "-a", "-b", fileName) ++ args
```
Comment (lefou) [inline scalalib/src/publish/SonatypePublisher.scala:157]: Wondering, if we should move these defaults into the default param to `gpgArgs` and document it's meaning. It's not a change request, just a thought. WDYT?

Comment (joan38) [reply scalalib/src/publish/SonatypePublisher.scala:157]: I was thinking the same but then people need to be aware that if there specify the `--gpgArgs` they need to provide these too to keep them.

Comment (lefou) [reply scalalib/src/publish/SonatypePublisher.scala:157]: We could add a hint in the scaladoc, too. But having these hard-coded args screams for future maintenance work, as seen in the past (#345, #530, #753). We could also rename the parameter to `alternativeGpgArgs` or `overrideGpgArgs` to indicate the fact that it replaced all defaults.

Comment (lefou) [reply scalalib/src/publish/SonatypePublisher.scala:157]: What about an accessor to the defaults to make reusing easy:

```scala
val defaultGpgArgs = ...
def publish(...,  gpgArgs: Seq[String] = defaultGpgArgs, ...)
```

And a simple example in the docs, showing how to add a passphrase using the defaultsArgs + the passphrase args

Comment (joan38) [reply scalalib/src/publish/SonatypePublisher.scala:157]: Sounds good to me. What do you think @jodersky?

Comment (jodersky) [reply scalalib/src/publish/SonatypePublisher.scala:157]: Looks good! I agree with @lefou's suggestion too. Thanks for working on this!

Comment (joan38) [reply scalalib/src/publish/SonatypePublisher.scala]: Change pushed

**Comment by lihaoyi** [comment]
This has broken publishing on the master branch:

```
Unknown arguments: "--gpgPassphrase" "[secure]"
```

@joan38 can you take a look?

**Comment by joan38** [comment]
@lihaoyi You need to pass `--gpgArgs '--passphrase [secure]'`

**Pull request merged**